### PR TITLE
CORE-1593 Mem Leak fix 🤞

### PR DESCRIPTION
**Pull request by Iapetus999**
- deleted some SLF4J artifacts that were causing issues in Vault
- Removed the Shade plugin since we only want a jar with LP code, not all the jars

### CODE REVIEW

File: src/main/java/org/littleshoot/proxy/impl/DefaultHttpProxyServer.java [hunk 1]
```diff
@@ -506,6 +506,14 @@ protected void registerChannel(Channel channel) {
         allChannels.add(channel);
     }
 
+    protected void unregisterChannel(Channel channel) {
+        if (channel.isOpen()) {
+            // Unlikely to happen but JIC
+            channel.closeFuture();
```
Comment (jentfoo) [inline src/main/java/org/littleshoot/proxy/impl/DefaultHttpProxyServer.java]: Based off the javadocs I believe this returns the future, but does not seem to start the close action: https://netty.io/4.1/api/io/netty/channel/Channel.html#closeFuture--

I believe you just want to invoke `close()`: https://netty.io/4.1/api/io/netty/channel/ChannelOutboundInvoker.html#close--

**Review by jentfoo — COMMENTED** [review]
Looks good in principle (minus the one comment left).  You may additionally want to consider reporting how many channels are open as a metric

**Comment by Iapetus999** [comment]
I was going to do that but as I was writing the code I discovered this issue. I'd rather keep the PR small and test this fix out, then if we still see leaks we can add metrics.

**Review by coanmj — APPROVED** [review]

File: src/main/java/org/littleshoot/proxy/Launcher.java [hunk 2]
```diff
@@ -134,10 +133,10 @@ private static void printHelp(final Options options,
     }
 
     private static void pollLog4JConfigurationFileIfAvailable() {
-        File log4jConfigurationFile = new File("src/test/resources/log4j.xml");
-        if (log4jConfigurationFile.exists()) {
-            DOMConfigurator.configureAndWatch(
-                    log4jConfigurationFile.getAbsolutePath(), 15);
-        }
+//        File log4jConfigurationFile = new File("src/test/resources/log4j.xml");
```
Comment (mjallday) [inline src/main/java/org/littleshoot/proxy/Launcher.java:136]: is this important?

**Review by mjallday — APPROVED** [review]

File: src/main/java/org/littleshoot/proxy/Launcher.java (hunk 2 above)
Comment (Iapetus999) [reply src/main/java/org/littleshoot/proxy/Launcher.java:136]: Only if it's run standalone. We prob can drop this file at some point.

**Pull request merged**